### PR TITLE
Document Playtime

### PR DESCRIPTION
**Pull request by Fexty12573**
Very small PR to document playtime.c + structs. Doesn't tie into many things other than trainercard display

### CODE REVIEW

File: include/struct_decls/playtime_decl.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#ifndef POKEPLATINUM_STRUCT_PLAYTIME_DECL_H
+#define POKEPLATINUM_STRUCT_PLAYTIME_DECL_H
+
+typedef struct Playtime Playtime;
```
Comment (adrienntindall) [inline include/struct_decls/playtime_decl.h:4]: You can prolly merge this into playtime.h

File: include/struct_defs/playtime.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#ifndef POKEPLATINUM_STRUCT_PLAYTIME_H
+#define POKEPLATINUM_STRUCT_PLAYTIME_H
+
+typedef struct Playtime {
```
Comment (adrienntindall) [inline include/struct_defs/playtime.h:4]: ditto - merge this into playtime.h

File: include/struct_defs/struct_02072014.h [hunk 3]
```diff
@@ -17,14 +17,14 @@ typedef struct {
     u8 unk_05;
     u8 padding_06[2];
     u16 unk_08[8];
-    const UnkStruct_02025E5C * unk_18;
+    const Playtime * unk_18;
```
Comment (adrienntindall) [inline include/struct_defs/struct_02072014.h:20]: name unk_18 -> playtime

File: include/unk_02017428.h [hunk 4]
```diff
@@ -1,10 +1,10 @@
 #ifndef POKEPLATINUM_UNK_02017428_H
 #define POKEPLATINUM_UNK_02017428_H
 
-#include "struct_decls/struct_02025E5C_decl.h"
+#include "struct_decls/playtime_decl.h"
 
 void sub_02017428(void);
-void sub_02017434(UnkStruct_02025E5C * param0);
+void sub_02017434(Playtime * param0);
```
Comment (adrienntindall) [inline include/unk_02017428.h:7]: param0 -> playtime

File: src/overlay005/ov5_021E1D20.c [hunk 5]
```diff
@@ -39,7 +39,7 @@ typedef struct {
     int unk_00;
     int unk_04;
     TrainerInfo * unk_08;
-    UnkStruct_02025E5C * unk_0C;
+    Playtime * unk_0C;
```
Comment (adrienntindall) [inline src/overlay005/ov5_021E1D20.c:42]: unk_0C -> playtime

File: src/playtime.c [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+#include "struct_defs/playtime.h"
+
+#include "playtime.h"
+
+#define PLAYTIME_MAX_HOURS 999
+#define PLAYTIME_MAX_MINUTES 59
+#define PLAYTIME_MAX_SECONDS 59
+
+void Playtime_Initialize(Playtime *playtime)
```
Comment (adrienntindall) [inline src/playtime.c:11]: we normally just shorten this to _Init

File: src/playtime.c [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+{
+    playtime->hours = 0;
+    playtime->minutes = 0;
+    playtime->seconds = 0;
+}
+
+void Playtime_Increase(Playtime *playtime, u32 seconds)
```
Comment (adrienntindall) [inline src/playtime.c:18]: Increment? Increase is probably fine.

File: include/struct_decls/playtime_decl.h (hunk 1 above)
Comment (adrienntindall) [reply include/struct_decls/playtime_decl.h:4]: also thoughts on PlayTime over Playtime? would also do playTime over playtime for variable names

File: src/unk_02017428.c [hunk 8]
```diff
@@ -1,24 +1,24 @@
...
 #include "unk_02022844.h"
-#include "unk_0202CBE4.h"
+#include "playtime.h"
 
 static BOOL Unk_021BF658 = 0;
 static u64 Unk_021BF668 = 0;
 static u64 Unk_021BF660 = 0;
 static u64 Unk_021BF670 = 0;
-static UnkStruct_02025E5C * Unk_021BF65C = NULL;
+static Playtime * Unk_021BF65C = NULL;
```
Comment (adrienntindall) [inline src/unk_02017428.c:14]: sPlaytime

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: include/struct_decls/playtime_decl.h (hunk 1 above)
Comment (lhearachel) [reply include/struct_decls/playtime_decl.h:4]: +1 on `PlayTime`

File: src/playtime.c (hunk 7 above)
Comment (lhearachel) [reply src/playtime.c:18]: Prefer `Increment`

File: include/playtime.h [hunk 9]
```diff
@@ -0,0 +1,16 @@
+#ifndef POKEPLATINUM_PLAYTIME_H
+#define POKEPLATINUM_PLAYTIME_H
+
+typedef struct PlayTime {
+    u16 hours;
+    u8 minutes;
+    u8 seconds;
+} PlayTime;
+
+void PlayTime_Init(PlayTime *playtime);
```
Comment (adrienntindall) [inline include/playtime.h:10]: playtime -> playTime

File: include/struct_defs/struct_0203E234.h [hunk 10]
```diff
@@ -1,14 +1,14 @@
 #ifndef POKEPLATINUM_STRUCT_0203E234_H
 #define POKEPLATINUM_STRUCT_0203E234_H
 
-#include "struct_decls/struct_02025E5C_decl.h"
+#include "playtime.h"
```
Comment (adrienntindall) [inline include/struct_defs/struct_0203E234.h:4]: thoughts on play_time.h?

File: include/struct_defs/struct_02072014.h [hunk 11]
```diff
@@ -17,14 +17,14 @@ typedef struct {
...
     u8 padding_06[2];
     u16 unk_08[8];
-    const UnkStruct_02025E5C * unk_18;
+    const PlayTime *playTime;
     u32 unk_1C;
     u32 unk_20;
     u32 unk_24;
     u16 unk_28;
-    u16 unk_2A;
+    u16 playtimeHours;
```
Comment (adrienntindall) [inline include/struct_defs/struct_02072014.h:25]: playTimeHours

File: include/struct_defs/struct_02072014.h [hunk 12]
```diff
@@ -17,14 +17,14 @@ typedef struct {
...
     u16 unk_2C;
-    u8 unk_2E;
+    u8 playtimeMinutes;
```
Comment (adrienntindall) [inline include/struct_defs/struct_02072014.h:27]: playTimeMinutes

File: src/overlay061/ov61_0222B008.c [hunk 13]
```diff
@@ -234,14 +234,14 @@ int ov61_0222B2B8 (UnkStruct_ov62_022349A8 * param0)
 
 int ov61_0222B2D8 (UnkStruct_ov62_022349A8 * param0, UnkStruct_02030A80 * param1, UnkStruct_ov61_0222B2D8 param2[])
 {
-    UnkStruct_02025E5C * v0;
+    PlayTime * v0;
```
Comment (adrienntindall) [inline src/overlay061/ov61_0222B008.c:237]: v0 -> playTime

File: src/unk_02025E08.c [hunk 14]
```diff
@@ -4,20 +4,18 @@
...
 #include "coins.h"
-#include "unk_0202CBE4.h"
+#include "playtime.h"
 
 struct UnkStruct_02025E0C_t {
     Options unk_00;
     TrainerInfo unk_04;
     u16 unk_24;
-    UnkStruct_02025E5C unk_26;
+    PlayTime playtime;
```
Comment (adrienntindall) [inline src/unk_02025E08.c:18]: playTime

File: src/unk_02025E08.c [hunk 15]
```diff
@@ -59,10 +57,8 @@ u16 * sub_02025E50 (SaveData * param0)
...
 }
 
-UnkStruct_02025E5C * sub_02025E5C (SaveData * param0)
+PlayTime *SaveData_GetPlayTime(SaveData *savedata)
 {
-    UnkStruct_02025E0C * v0;
-
-    v0 = SaveData_SaveTable(param0, 1);
-    return &v0->unk_26;
+    UnkStruct_02025E0C *player = SaveData_SaveTable(savedata, 1);
```
Comment (adrienntindall) [inline src/unk_02025E08.c:62]: why'd you label this player?

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: src/unk_02025E08.c (hunk 15 above)
Comment (Fexty12573) [reply src/unk_02025E08.c:62]: I figured it made sense but I will change it back

File: src/overlay005/ov5_021E1D20.c [hunk 16]
```diff
@@ -4,7 +4,7 @@
 #include "message.h"
 #include "struct_decls/struct_02018340_decl.h"
 #include "strbuf.h"
-#include "struct_decls/struct_02025E5C_decl.h"
+#include "play_time.h"
```
Comment (lhearachel) [inline src/overlay005/ov5_021E1D20.c:7]: Extraneous include

File: src/overlay097/ov97_0222AE60.c [hunk 17]
```diff
@@ -8,7 +8,7 @@
 #include "struct_decls/struct_02022550_decl.h"
 #include "strbuf.h"
 #include "struct_decls/struct_0202442C_decl.h"
-#include "struct_decls/struct_02025E5C_decl.h"
+#include "play_time.h"
```
Comment (lhearachel) [inline src/overlay097/ov97_0222AE60.c:11]: Extraneous include

File: src/unk_02025E08.c [hunk 18]
```diff
@@ -59,10 +57,8 @@ u16 * sub_02025E50 (SaveData * param0)
...
 
-UnkStruct_02025E5C * sub_02025E5C (SaveData * param0)
+PlayTime *SaveData_GetPlayTime(SaveData *savedata)
 {
-    UnkStruct_02025E0C * v0;
-
-    v0 = SaveData_SaveTable(param0, 1);
-    return &v0->unk_26;
+    UnkStruct_02025E0C *v0 = SaveData_SaveTable(savedata, 1);
+    return &v0->playTime;
```
Comment (lhearachel) [inline src/unk_02025E08.c:63]: Does this match?

```c
return &SaveData_SaveTable(savedata, 1)->playTime;
```

File: src/play_time.c [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+#define PLAYTIME_MAX_HOURS 999
+#define PLAYTIME_MAX_MINUTES 59
+#define PLAYTIME_MAX_SECONDS 59
+
+void PlayTime_Init(PlayTime *playTime)
+{
+    playTime->hours = 0;
+    playTime->minutes = 0;
+    playTime->seconds = 0;
+}
```
Comment (lhearachel) [inline src/play_time.c:14]: Does this match as `memset(playTime, 0, sizeof(PlayTime))`, per-chance?

File: src/play_time.c [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+        return;
+    }
+
+    u32 hours = playTime->hours;
+    u32 minutes = playTime->minutes;
+    u32 newSeconds = playTime->seconds + seconds;
+
+    if (newSeconds > PLAYTIME_MAX_SECONDS) {
+        minutes = playTime->minutes + newSeconds / (PLAYTIME_MAX_SECONDS + 1);
+        newSeconds %= PLAYTIME_MAX_SECONDS + 1;
```
Comment (lhearachel) [inline src/play_time.c:30]: Do these deserve their own consts for readability?

```c
#define SEC_PER_MIN  60
#define MIN_PER_HOUR 60
```

File: src/unk_02025E08.c (hunk 18 above)
Comment (Fexty12573) [reply src/unk_02025E08.c:63]: You missing a cast there because `SaveData_SaveTable` returns `void*`, this here does match
```c
return &((UnkStruct_02025E0C *)SaveData_SaveTable(savedata, 1))->playTime;
```
I honestly wouldn't really say this looks nicer tho..

File: src/play_time.c (hunk 19 above)
Comment (Fexty12573) [reply src/play_time.c:14]: It does not

File: src/unk_02025E08.c (hunk 18 above)
Comment (lhearachel) [reply src/unk_02025E08.c:63]: I agree, that looks way worse. Keep it as is.

**Review by lhearachel — APPROVED** [review]
